### PR TITLE
[Merged by Bors] - chore: various style improvements in positivity extensions

### PR DESCRIPTION
**Pull request by dwrensha**
- match with ~q()
- put type ascriptions inline

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/Positivity/Basic.lean [hunk 1]
```diff
@@ -437,7 +429,8 @@ def evalNatCast : PositivityExt where eval {u α} _zα _pα e := do
 
 @[positivity Int.cast _]
 def evalIntCast : PositivityExt where eval {u α} _zα _pα e := do
-  let (.app _ (a : Q(Int))) ← withReducible (whnf e) | throwError "not Int.cast"
+  let _iα : Q(IntCast $α) ← synthInstanceQ q(IntCast $α)
+  let ~q(Int.cast ($a : ℤ)) := e | throwError "not Int.cast"
```
Comment (eric-wieser) [inline Mathlib/Tactic/Positivity/Basic.lean:433]: Arguably using `@` would be better here, as that saves a TC search.

Comment (dwrensha) [reply Mathlib/Tactic/Positivity/Basic.lean]: Done in https://github.com/leanprover-community/mathlib4/pull/8098/commits/3a2cfb36b1104db495ed4517267814f0f68a9f16

Comment (kmill) [reply Mathlib/Tactic/Positivity/Basic.lean]: Are we worried about `Int.cast` having the wrong `IntCast` instance? It looks like it incidentally was checking that the `Int.cast` expression had the canonical one.

File: Mathlib/Tactic/Positivity/Basic.lean [hunk 2]
```diff
@@ -381,8 +373,8 @@ private theorem abs_pos_of_ne_zero {α : Type*} [AddGroup α] [LinearOrder α]
 
 /-- The `positivity` extension which identifies expressions of the form `|a|`. -/
 @[positivity |(_ : α)|]
-def evalAbs : PositivityExt where eval {_ _α} zα pα e := do
-  let (.app _ (a : Q($_α))) ← withReducible (whnf e) | throwError "not |·|"
+def evalAbs : PositivityExt where eval {u} (α : Q(Type u)) zα pα (e : Q($α)) := do
+  let ~q(@Abs.abs _ $_aα $a) := e | throwError "not |·|"
```
Comment (eric-wieser) [inline Mathlib/Tactic/Positivity/Basic.lean:377]:
```suggestion
  let ~q(@Abs.abs _ (_) $a) := e | throwError "not |·|"
```
to match the others

**Comment by robertylewis** [comment]
bors merge

**Pull request closed without merging**